weblab/experiments/tests/test_views.py

```diff
@@ -892,6 +897,93 @@ def test_view_experiment_version(self, client, experiment_version):
+        assert response.status_code == 302
+
+    @pytest.mark.usefixtures('logged_in_user')
+    def test_load_page_logged_in_user(self, client):
+        response = client.get('/experiments/tasks')
+        assert response.status_code == 200
+
+    @pytest.mark.usefixtures('logged_in_user')
+    def test_get_queryset_other_user(self, other_user, client, experiment_version):
+        experiment_version.author = other_user
+        experiment_version.save()
+        recipes.running_experiment.make(experiment_version=experiment_version)
+        assert RunningExperiment.objects.count() == 1
+        response = client.get('/experiments/tasks')
+        assert len(response.context['runningexperiment_list']) == 0
+
+    def test_get_queryset(self, logged_in_user, client, helpers):
+        # Create three experiment versions 2 running and 1 completed
+        model_1 = recipes.model.make(author=logged_in_user)
+        model_1_version = helpers.add_version(model_1, visibility='public')
+        protocol_1 = recipes.protocol.make(author=logged_in_user)
+        protocol_1_version = helpers.add_version(protocol_1, visibility='public')
+        protocol_1_version2 = helpers.add_version(protocol_1, visibility='public')
+        protocol_2 = recipes.protocol.make(author=logged_in_user)
+        protocol_2_version = helpers.add_version(protocol_2, visibility='public')
+
+        recipes.experiment_version.make(
+            status=ExperimentVersion.STATUS_SUCCESS,
+            experiment__model=model_1,
+            experiment__model_version=model_1_version.hexsha,
+            experiment__protocol=protocol_1,
+            experiment__protocol_version=protocol_1_version.hexsha,
+            author=logged_in_user,
+        )
+
+        exp_version_2 = recipes.experiment_version.make(
+            status=ExperimentVersion.STATUS_RUNNING,
+            experiment__model=model_1,
+            experiment__model_version=model_1_version.hexsha,
+            experiment__protocol=protocol_1,
+            experiment__protocol_version=protocol_1_version2.hexsha,
+            author=logged_in_user,
+        )
+        running_exp_version2 = recipes.running_experiment.make(experiment_version=exp_version_2)
+
+        exp_version_3 = recipes.experiment_version.make(
+            status=ExperimentVersion.STATUS_RUNNING,
+            experiment__model=model_1,
+            experiment__model_version=model_1_version.hexsha,
+            experiment__protocol=protocol_2,
+            experiment__protocol_version=protocol_2_version.hexsha,
+            author=logged_in_user,
+        )
+        running_exp_version3 = recipes.running_experiment.make(experiment_version=exp_version_3)
+
+        assert ExperimentVersion.objects.count() == 3
+        assert RunningExperiment.objects.count() == 2
+
+        # Return only running experiment versions
+        response = client.get('/experiments/tasks')
+        assert len(response.context['runningexperiment_list']) == 2
```

Comment:
I wouldn't check both len and the set contents - checking the set contents will do both, and also give a more useful error message if the test fails (since you'll see which items didn't match).

Reply:
agreed